vespa-http-client/src/main/java/com/yahoo/vespa/http/client/core/communication/IOThread.java

```diff
@@ -595,12 +599,9 @@ private boolean timeToPoll(GatewayConnection connection) {
             if (connection.lastPollTime() == null) return true;
 
             // Exponential (2^x) dropoff:
-            double unit = pollIntervalUS / 1000.0;
-            double x = (  clock.millis() - connection.connectionTime().plus(connectionTimeToLive).toEpochMilli() ) / unit;
-            double lastX = (  connection.lastPollTime().toEpochMilli() - connection.connectionTime().plus(connectionTimeToLive).toEpochMilli() ) / unit;
-
-            double currentDelayDoublings = Math.log(lastX)/Math.log(2);
-            return (x - lastX) > Math.pow(2, currentDelayDoublings);
+            double connectionEndOfLife = connection.connectionTime().plus(connectionTimeToLive).toEpochMilli();
+            double connectionLastPolled = connection.lastPollTime().toEpochMilli();
+            return clock.millis() - connectionEndOfLife > 2 * (connectionLastPolled - connectionEndOfLife);
```

Comment:
Thanks for dissolving my math! I just wrote down the function I had in mind :stuck_out_tongue_closed_eyes:

Reply:
😜